diff --git a/var/spack/repos/builtin/packages/matlab/package.py b/var/spack/repos/builtin/packages/matlab/package.py
index 249213ac..449e3f3d 100644
--- a/var/spack/repos/builtin/packages/matlab/package.py
+++ b/var/spack/repos/builtin/packages/matlab/package.py
@@ -26,6 +26,7 @@ class Matlab(Package):
     homepage = "https://www.mathworks.com/products/matlab.html"
     manual_download = True

+    version("R2024a", sha256="d62123ebef6378a3f9c09dc5d92dd8220df091e3a7d4559fc918c698a05edcda")
     version("R2019b", sha256="d60787263afb810283b7820c4c8d9cb1f854c7cb80f47e136643fd95bf5fbd59")
     version("R2018b", sha256="8cfcddd3878d3a69371c4e838773bcabf12aaf0362cc2e1ae7e8820845635cac")
     version("R2016b", sha256="a3121057b1905b132e5741de9f7f8350378592d84c5525faf3ec571620a336f2")
@@ -52,7 +53,7 @@ class Matlab(Package):
     extendable = True

     def url_for_version(self, version):
-        return "file://{0}/matlab_{1}_glnxa64.zip".format(os.getcwd(), version)
+        return "matlab_{0}_glnxa64.zip".format(version)

     def install(self, spec, prefix):
         config = {
@@ -73,15 +74,3 @@ def install(self, spec, prefix):
         input_file = join_path(self.stage.source_path, "spack_installer_input.txt")
         subprocess.call(["./install", "-inputFile", input_file])
-
-    @run_after("install")
-    def post_install(self):
-        # Fix broken link
-        with working_dir(self.spec.prefix.bin.glnxa64):
-            os.unlink("libSDL2.so")
-            os.symlink("libSDL2-2.0.so.0.2.1", "libSDL2.so")
-
-        # Fix to random exceptions when changing display settings
-        # https://www.mathworks.com/matlabcentral/answers/373897-external-monitor-throws-java-exception
-        java_opts = os.path.join(self.spec.prefix.bin.glnxa64, "java.opts")
-        with open(java_opts, "w") as out:
-            out.write("-Dsun.java2d.xrender=false\n")
+
